src/componenets/signinPage/index.jsx

```diff
@@ -1,29 +1,69 @@
-import React from "react";
+import React, { useState } from "react";
 import { Box, Container, Card, Typography, Grid } from "@mui/material";
 import CustomButton from "../../common/button";
 import { Link } from "react-router-dom";
 import CustomTextField from "../../common/textField/index";
+import { useNavigate } from "react-router-dom";
+import { signIn } from "../../services/dataServices";
+import toast from "react-hot-toast";
+import { Toaster } from "react-hot-toast";
+import { useDispatch } from "react-redux";
+import { addUser } from "../../redux/actions/userAction";
 
 const SigninPage = () => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [loading, setLoading] = useState(false);
+  const dispatch = useDispatch();
+
+  const navigate = useNavigate();
+  const path = "/";
+
+  const onSubmit = async (e) => {
+    e.preventDefault();
+    setLoading(true);
+    const payload = {
+      email: email,
+      password: password,
+    };
+
+    const result = await signIn(payload);
+    if (result.status === 200) {
+      localStorage.setItem("token", result.data.payload.data.token);
+      toast.success(result.data.metadata.message);
+
+      setTimeout(() => {
+        navigate(path);
+      }, 2000);
```

Comment:
please don't add setTimout
and don't use toast in success case 
it doesn't matter 
navigation is itself indication of success